spec/unit/pops/lookup/context_spec.rb

```diff
@@ -171,13 +171,8 @@ module Lookup
 
       context 'and multiple compilations' do
 
-        before(:each) { Puppet.settings[:environment_timeout] = 'unlimited' }
-        after(:each) do
-          Puppet.settings[:environment_timeout] = 0
-          Puppet.lookup(:environments).clear_all
-        end
-
         it 'will reuse cached_file_data and not call block again' do
+          Puppet.settings[:environment_timeout] = 'unlimited'
```

Comment:
Ok.. I had a feeling that my modification invalidated what you were trying to assert - but the code wasn't totally clear to me.  I also didn't realize that the file change reload and the environment reload were separate circumstances.  I thought the test might have been generating a false positive success for you because you were running the specs differently - but my assumption there seems incorrect.

Are you running on OSX?  This test failed for @joshcooper, @MosesMendoza and myself locally, which is why we were making adjustments. We're all running OSX - not sure if you're running Linux (the test seemingly passes in TravisCI in its prior form)?